### PR TITLE
[KYUUBI #2565] Variable substitution should work in plan only mode

### LINKED ISSUE #2565: [Bug] Variable substitution does not work in plan only mode
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

```log
0: jdbc:hive2://192.168.208.105:10009/> set kyuubi.operation.plan.only.mode=parse;
10:21:15.802 INFO org.apache.kyuubi.operation.ExecuteStatement: Processing fchen's query[5027f6c0-03bd-4541-861c-ef12da759ad4]: INITIALIZED_STATE -> PENDING_STATE, statement: set kyuubi.operation.plan.only.mode=parse
10:21:15.846 INFO org.apache.kyuubi.operation.ExecuteStatement: Processing fchen's query[5027f6c0-03bd-4541-861c-ef12da759ad4]: PENDING_STATE -> RUNNING_STATE, statement: set kyuubi.operation.plan.only.mode=parse
10:21:15.835 INFO org.apache.kyuubi.engine.spark.operation.ExecuteStatement: Processing fchen's query[b4e7c9d5-9a47-4185-8a0b-b4ad9655dc13]: INITIALIZED_STATE -> PENDING_STATE, statement: set kyuubi.operation.plan.only.mode=parse
10:21:15.839 INFO org.apache.kyuubi.engine.spark.operation.ExecuteStatement: Processing fchen's query[b4e7c9d5-9a47-4185-8a0b-b4ad9655dc13]: PENDING_STATE -> RUNNING_STATE, statement: set kyuubi.operation.plan.only.mode=parse
10:21:15.841 INFO org.apache.kyuubi.engine.spark.operation.ExecuteStatement:
           Spark application name: kyuubi_USER_SPARK_SQL_fchen_default_6c5b23b2-e017-48c1-9f30-7bdee7f42815
                 application ID: local-1651803664826
                 application web UI: http://192.168.208.105:50619
                 master: local[*]
                 deploy mode: client
                 version: 3.2.0
           Start time: 2022-05-06T10:21:03.817
           User: fchen
10:21:15.876 INFO org.apache.kyuubi.engine.spark.operation.ExecuteStatement: Execute in full collect mode
10:21:15.896 INFO org.apache.spark.sql.catalyst.expressions.codegen.CodeGenerator: Code generated in 9.750672 ms
10:21:15.927 INFO org.apache.spark.sql.catalyst.expressions.codegen.CodeGenerator: Code generated in 9.754239 ms
10:21:15.933 INFO org.apache.kyuubi.engine.spark.operation.ExecuteStatement: Processing fchen's query[b4e7c9d5-9a47-4185-8a0b-b4ad9655dc13]: RUNNING_STATE -> RUNNING_STATE, statement: set kyuubi.operation.plan.only.mode=parse
10:21:15.933 INFO org.apache.kyuubi.engine.spark.operation.ExecuteStatement: Processing fchen's query[b4e7c9d5-9a47-4185-8a0b-b4ad9655dc13]: RUNNING_STATE -> FINISHED_STATE, statement: set kyuubi.operation.plan.only.mode=parse, time taken: 0.094 seconds
10:21:15.983 INFO org.apache.kyuubi.operation.ExecuteStatement: Query[5027f6c0-03bd-4541-861c-ef12da759ad4] in FINISHED_STATE
10:21:15.983 INFO org.apache.kyuubi.operation.ExecuteStatement: Processing fchen's query[5027f6c0-03bd-4541-861c-ef12da759ad4]: RUNNING_STATE -> FINISHED_STATE, statement: set kyuubi.operation.plan.only.mode=parse, time taken: 0.137 seconds
+----------------------------------+--------+
|               key                | value  |
+----------------------------------+--------+
| kyuubi.operation.plan.only.mode  | parse  |
+----------------------------------+--------+
1 row selected (8.99 seconds)
0: jdbc:hive2://192.168.208.105:10009/> set x=y;
10:21:24.797 INFO org.apache.kyuubi.operation.ExecuteStatement: Processing fchen's query[9cd03fd0-a825-4f16-8412-14795c9e3f09]: INITIALIZED_STATE -> PENDING_STATE, statement: set x=y
10:21:24.834 INFO org.apache.kyuubi.operation.ExecuteStatement: Processing fchen's query[9cd03fd0-a825-4f16-8412-14795c9e3f09]: PENDING_STATE -> RUNNING_STATE, statement: set x=y
10:21:24.850 INFO org.apache.kyuubi.operation.ExecuteStatement: Query[9cd03fd0-a825-4f16-8412-14795c9e3f09] in FINISHED_STATE
10:21:24.850 INFO org.apache.kyuubi.operation.ExecuteStatement: Processing fchen's query[9cd03fd0-a825-4f16-8412-14795c9e3f09]: RUNNING_STATE -> FINISHED_STATE, statement: set x=y, time taken: 0.016 seconds
+------+--------+
| key  | value  |
+------+--------+
| x    | y      |
+------+--------+
1 row selected (0.256 seconds)
0: jdbc:hive2://192.168.208.105:10009/> select '${x}';
10:21:30.636 INFO org.apache.kyuubi.operation.ExecuteStatement: Processing fchen's query[65402727-35c6-4692-ab38-6bf4343b45dd]: INITIALIZED_STATE -> PENDING_STATE, statement: select '${x}'
10:21:30.690 INFO org.apache.kyuubi.operation.ExecuteStatement: Processing fchen's query[65402727-35c6-4692-ab38-6bf4343b45dd]: PENDING_STATE -> RUNNING_STATE, statement: select '${x}'
10:21:30.694 INFO org.apache.kyuubi.operation.ExecuteStatement: Query[65402727-35c6-4692-ab38-6bf4343b45dd] in FINISHED_STATE
10:21:30.694 INFO org.apache.kyuubi.operation.ExecuteStatement: Processing fchen's query[65402727-35c6-4692-ab38-6bf4343b45dd]: RUNNING_STATE -> FINISHED_STATE, statement: select '${x}', time taken: 0.004 seconds
+----------------------------------------------------+
|                        plan                        |
+----------------------------------------------------+
| 'Project [unresolvedalias(, None)]
+- OneRowRelation
 |
+----------------------------------------------------+
1 row selected (0.075 seconds)

```

### Affects Version(s)

master

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by cfmcgrady**
### _Why are the changes needed?_

close #2565 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by cfmcgrady** [comment]
cc @yaooqinn

**Comment by ulysses-you** [comment]
good cache ! I looked at the `SparkSQLDriver`, it seems with conf is enough
```scala
val substitutorCommand = SQLConf.withExistingConf(context.conf) {
  new VariableSubstitution().substitute(command)
}
context.sparkContext.setJobDescription(substitutorCommand)
```

we can also update it for `setJobDescription`

**Comment by gabrywu** [comment]
according to https://github.com/apache/incubator-kyuubi/issues/1059, engine should not run `set x = y`, right ?

**Comment by ulysses-you** [comment]
@gabrywu  it depends on the config `kyuubi.operation.plan.only.excludes`. By default, we allow running `set`.

**Comment by cfmcgrady** [comment]
> we can also update it for `setJobDescription`

Since the construction of `VariableSubstitution` is different across multiple versions of Spark, set the job description to the unsubstituted statement only.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/PlanOnlyStatement.scala [hunk 1]
```diff
@@ -52,30 +53,33 @@ class PlanOnlyStatement(
...
-              val optimized = spark.sessionState.optimizer.execute(analyzed)
-              iter = new IterableFetchIterator(Seq(Row(optimized.toString())))
-            case PHYSICAL =>
-              val physical = spark.sql(statement).queryExecution.sparkPlan
-              iter = new IterableFetchIterator(Seq(Row(physical.toString())))
-            case EXECUTION =>
-              val executed = spark.sql(statement).queryExecution.executedPlan
-              iter = new IterableFetchIterator(Seq(Row(executed.toString())))
-          }
+      spark.sparkContext.setJobDescription(statement)
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/PlanOnlyStatement.scala:56]: use `withLocalProperties` like `ExecuteStatement` ?

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**